addons/uom/static/src/js/uom_field_widget.js

```diff
@@ -0,0 +1,151 @@
+    const utils = require('web.utils');
+
+    const _t = core._t;
+    const FieldFloat = basic_fields.FieldFloat;
+    const InputField = basic_fields.InputField;
+
+    const FieldMeasure = FieldFloat.extend({
+        supportedFieldTypes: ['float', 'measure'],
+        resetOnAnyFieldChange: true,  // TODO: only listen to uom field's change, but how?
+
+        events: Object.assign({}, InputField.prototype.events, {
+            input: '_onInput',
+        }),
+
+        /**
+         * Float fields using a measure widget have an additional uom_field
+         * parameter which defines the name of the field from which the uom
+         * should be read.
+         *
+         * If no uom field is given or the field does not exist, we fallback
+         * to the default input behavior instead.
+         *
+         * @override
+         */
+        init: function () {
+            this._super(...arguments);
+            this._setUom();
+        },
+
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+        /**
+         * Re-gets the uom as its value may have changed.
+         * @see FieldUom.resetOnAnyFieldChange
+         *
+         * @override
+         * @private
+         */
+        _reset: function () {
+            this._super(...arguments);
+            this._setUom();
+        },
+
+        /**
+         * Deduces the uom description from the field options and view state.
+         * The description is then available at this.uom.
+         *
+         * @private
+         */
+        _setUom: function () {
+            const uomField = this.nodeOptions.uom_field || this.field.uom_field || 'uom_id' || 'x_uom_id';
+            const uomID = this.record.data[uomField] && this.record.data[uomField].res_id;
+            this.uom = session.uoms[uomID];
+            this.formatOptions.uom = this.uom; // _formatValue() uses formatOptions
+        },
+
+        //--------------------------------------------------------------------------
+        // Handlers
+        //--------------------------------------------------------------------------
+        /**
+         * Allow to enter decimal digits base on the uom's rounding value.
+         *
+         * @override
+         * @private
+         */
+        _onInput: function() {
+            this._super(...arguments);
+            const value = this.$input.val();
+            const uom_precision = this.formatOptions.uom && this.formatOptions.uom.decimal_places;
+            if (value.includes('.')) {
+                const precision = value.split('.')[1].length;
+                if (uom_precision < precision) {
+                    this.do_warn(_.str.sprintf(_t('Only %s decimals are allowed.</br> This setting can be configured on the Unit of Measure.'), uom_precision));
+                    this.$input.val(value.slice(0,-1));
```

Comment:
Why `value.slice(0,-1)` ? This assumes the `value` will be only one char too long, isn't ?
It should be dependent of the difference between `precision` and `uom_precision` I think:
```suggestion
                    this.$input.val(value.slice(0, -(precision - uom_precision)));
```